[C#] Base64Encode does not exist in the current context

Revision: 62fbcb9e8bec4a40569e51c7f316af85806628ee

I'm getting this error in generated csharp clients:

```
error CS0103: The name `Base64Encode' does not exist in the current context
```

The issue appears to be `Base64Encode` being called within a scope not including the method. I've fixed it by adding `ApiClient.` to all the call-sites. See https://github.com/DocRaptor/docraptor-csharp/commit/f2e33e6697f4fd02278e2f691cc684ea08a58dba. I'm not a C# programmer so I'm not entirely sure this is the right fix, but it does work for me :)

Anyhow, I see no way for this code to work as-is. It seems like we must be missing a test case for the HTTP authentication in C#.

--- a/modules/swagger-codegen/src/main/resources/csharp/api.mustache
+++ b/modules/swagger-codegen/src/main/resources/csharp/api.mustache
@@ -228,7 +228,7 @@ namespace {{packageName}}.Api
             // http basic authentication required
             if (!String.IsNullOrEmpty(Configuration.Username) || !String.IsNullOrEmpty(Configuration.Password))
             {
-                localVarHeaderParams["Authorization"] = "Basic " + Base64Encode(Configuration.Username + ":" + Configuration.Password);
+                localVarHeaderParams["Authorization"] = "Basic " + ApiClient.Base64Encode(Configuration.Username + ":" + Configuration.Password);
             }{{/isBasic}}{{#isOAuth}}
             // oauth required
             if (!String.IsNullOrEmpty(Configuration.AccessToken))
@@ -340,7 +340,7 @@ namespace {{packageName}}.Api
             // http basic authentication required
             if (!String.IsNullOrEmpty(Configuration.Username) || !String.IsNullOrEmpty(Configuration.Password))
             {
-                localVarHeaderParams["Authorization"] = "Basic " + Base64Encode(Configuration.Username + ":" + Configuration.Password);
+                localVarHeaderParams["Authorization"] = "Basic " + ApiClient.Base64Encode(Configuration.Username + ":" + Configuration.Password);
             }{{/isBasic}}{{#isOAuth}}
             // oauth required
             if (!String.IsNullOrEmpty(Configuration.AccessToken))

--- a/modules/swagger-codegen/src/test/resources/2_0/petstore.json
+++ b/modules/swagger-codegen/src/test/resources/2_0/petstore.json
@@ -1091,7 +1091,12 @@
           "400": {
             "description": "Invalid username supplied"
           }
-        }
+        },
+        "security": [
+          {
+            "test_http_basic": []
+          }
+        ]
       }
     }
   },
@@ -1129,6 +1134,9 @@
       "type": "apiKey",
       "name": "test_api_key_query",
       "in": "query"
+    },
+    "test_http_basic": {
+      "type": "basic"
     }
   },
   "definitions": {

--- a/samples/client/petstore/csharp/SwaggerClientTest/Lib/SwaggerClient/src/main/csharp/IO/Swagger/Api/UserApi.cs
+++ b/samples/client/petstore/csharp/SwaggerClientTest/Lib/SwaggerClient/src/main/csharp/IO/Swagger/Api/UserApi.cs
@@ -1001,6 +1001,13 @@ namespace IO.Swagger.Api
             
             
 
+            // authentication (test_http_basic) required
+            
+            // http basic authentication required
+            if (!String.IsNullOrEmpty(Configuration.Username) || !String.IsNullOrEmpty(Configuration.Password))
+            {
+                localVarHeaderParams["Authorization"] = "Basic " + ApiClient.Base64Encode(Configuration.Username + ":" + Configuration.Password);
+            }
             
     
             // make the HTTP request
@@ -1079,6 +1086,14 @@ namespace IO.Swagger.Api
             
             
 
+            
+            // authentication (test_http_basic) required
+            
+            // http basic authentication required
+            if (!String.IsNullOrEmpty(Configuration.Username) || !String.IsNullOrEmpty(Configuration.Password))
+            {
+                localVarHeaderParams["Authorization"] = "Basic " + ApiClient.Base64Encode(Configuration.Username + ":" + Configuration.Password);
+            }
             
 
             // make the HTTP request

